### PR TITLE
Daily challenges UI. Closes #32

### LINKED ISSUE #32: Daily Challenges Page UI
- Need to implement @kelpikz sir's [design](https://www.figma.com/file/OtcEA7dW5XD31ex5sO5aDB/Mobile?node-id=613%3A385)
- And complete the todos in #27

### PR DESCRIPTION
**Pull request by Sudhindra3000**
Features:
- UI for Daily challenges page
- Realtime updates in progress of current market day challenges
- Retry button in Splash page when state is `DalalLoginFailed`

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -73,7 +72,7 @@ class DalalApp extends StatelessWidget {
               if (state.user.isPhoneVerified) {
                 //showSnackBar(context, 'Welcome ${state.user.name}');
                 _navigator.pushNamedAndRemoveUntil(
-                  '/home',
+                  '/dailyChallenges',
```
Comment (dhruv1294) [inline lib/main.dart:75]: change it back to /home

File: lib/theme/colors.dart [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import 'package:flutter/material.dart';
 
 const primaryColor = Color(0xff23E464);
-const backgroundColor = Color(0xff161B1B);
+const backgroundColor = Color(0xff030407);
```
Comment (dhruv1294) [inline lib/theme/colors.dart:4]: this will change in all pages since all have used this color :-)

Comment (Sudhindra3000) [reply lib/theme/colors.dart:4]: Ya there is a different background color for auth pages and for main pages. We should handle that

File: lib/main.dart (hunk 1 above)
Comment (Sudhindra3000) [reply lib/main.dart:75]: Ok

**Review by dhruv1294 — APPROVED** [review]

File: lib/blocs/daily_challenges/challenge_reward.dart/challenge_reward_state.dart [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  const ChallengeRewardLoading();
+}
+
+class ChallengeRewardFailure extends ChallengeRewardState {
+  final String msg;
+
+  const ChallengeRewardFailure(this.msg);
+}
+
+class ChallengeRewardCalimed extends ChallengeRewardState {
```
Comment (0149Sailesh) [inline lib/blocs/daily_challenges/challenge_reward.dart/challenge_reward_state.dart:28]: ~Calimed~ Claimed 🙂

Comment (Sudhindra3000) [reply lib/blocs/daily_challenges/challenge_reward.dart/challenge_reward_state.dart:28]: Ok 👍

**Comment by 0149Sailesh** [comment]
Are we adding daily challenges notification along with this PR?

**Review by 0149Sailesh — APPROVED** [review]

**Pull request merged**